cinch/roles/jenkins_master/tasks/configure.yml

```diff
@@ -23,5 +23,18 @@
     group: jenkins
     state: directory
 
+- name: set hostname to match inventory hostname
+  hostname:
+    name: "{{ inventory_hostname }}"
+  when: jenkins_ssl_cert is not defined
```

Comment:
So inventory_hostname is not the appropriate value to set here, as that name does not necessarily have to correspond to a value that is even a valid hostname. The better solution is to modify the ensure_up file to work with potentially invalid certificates. Preferably by setting a value like "validate_certs: false" on its uri call.